sdk/java/src/main/java/sawtooth/sdk/processor/TransactionProcessor.java

```diff
@@ -60,64 +103,109 @@ public void addHandler(TransactionHandler handler) {
             .setEncoding(handler.getEncoding())
             .setVersion(handler.getVersion())
             .build();
-    this.stream.send(Message.MessageType.TP_REGISTER_REQUEST, registerRequest.toByteString());
-    this.handlers.add(handler);
+    try {
+      Future fut = this.stream.send(
+          Message.MessageType.TP_REGISTER_REQUEST, registerRequest.toByteString());
+      fut.getResult();
+      this.registered = true;
+      this.handlers.add(handler);
+    } catch (InterruptedException ie) {
+      ie.printStackTrace();
+    } catch (ValidatorConnectionError vce) {
+      logger.info(vce.toString());
+    }
   }
 
-
-  public void stopRunning() {
-    this.isRunning = false;
+  /**
+  * Get the current message that is being processed.
+  */
+  private Message getCurrentMessage() {
+    return this.currentMessage;
   }
 
+  /**
+  * Used to process a message.
+  * @param message The Message to process.
+  * @param stream The Stream to use to send back responses.
+  * @param handler The handler that should be used to process the message.
+  */
+  private static void process(Message message, Stream stream,
+      TransactionHandler handler) {
+    try {
+      TpProcessRequest transactionRequest = TpProcessRequest
+              .parseFrom(message.getContent());
+      State state = new State(stream, transactionRequest.getContextId());
+      try {
+        handler.apply(transactionRequest, state);
+        TpProcessResponse response = TpProcessResponse.newBuilder()
+                .setStatus(TpProcessResponse.Status.OK).build();
+
+        stream.sendBack(Message.MessageType.TP_PROCESS_RESPONSE,
+                message.getCorrelationId(),
+                response.toByteString());
+      } catch (InvalidTransactionException ite) {
+        ite.printStackTrace();
```

Comment:
Let's not log this  stack trace.  Instead, log the message of the Transaction as part of the warning below.